### DISCUSSION TITLE
Is there a way to pack mipmaps into one file, using cmgen or mipgen?

### DISCUSSION THREAD

**Discussion by EnviralDesign** [google/filament#5103] [Q&A]
Hello, I'm using the cmgen program as a command line utility, to generate the prefiltered ibl. the result is many files, and since I'm using this and the filament shader system in another project, it's very tedious to manually compile all these mip levels and cube map faces into a single atlas.

I have a solution to automate the assembly of the cubemap faces, but I have not yet found a way to merge all the mip levels into one texture, like DDS, that supports mip maps natively.

Any idea on how to do this, inside or outside of Filament?

Right now, this is my batch file:

```
"C:\tmp\filament\bin\cmgen.exe" ^
--size=256 ^
--deploy="C:\tmp\filament" ^
--format=dds ^
--extract-blur=0.1 ^
--ibl-dfg="dfg.exr" ^
--ibl-dfg-multiscatter ^
--ibl-dfg-cloth ^
--sh-output="sh.exr" ^
"C:\tmp\filament\venetian_crossroads_2k.hdr"
```

**Comment by romainguy** [comment]
You can use -f ktx to ask cmgen to export results as a KTX file containing all mip levels and all cube faces.

**Reply by EnviralDesign** [reply]
Ahh, is ktx the only format that supports embedded mip levels / cube faces? or is it possible to also do the same with dds or similar?
The gfx package I'm bringing these into doesn't at present support ktx, but I could bring that up to the developers if ktx is the only way.

**Reply by pixelflinger** [reply]
Yes we only support `.ktx`.
